### DISCUSSION TITLE
Running two parallel delayed processes

### DISCUSSION THREAD

**Discussion by liadJB** [bonsai-rx/bonsai#1720] [Q&A]
Hi, 
I am attaching the script I am using:
[RewardImagingDelayed.zip](https://github.com/bonsai-rx/bonsai/files/14758648/RewardImagingDelayed.zip)

1) I have a part that sends a pulse that gives a reward to the mouse:
![image](https://github.com/bonsai-rx/bonsai/assets/48135253/88440157-41c9-49c5-bf27-408d25f8953d)
I want this to start running several seconds into the script start. I tried various ways: encapsulating it into an async or an observable and attach a delay node - none of which worked.

2) Another problem is that I have this part, that sends a pulse that starts the recording
![image](https://github.com/bonsai-rx/bonsai/assets/48135253/d0f16fd1-c683-4862-bcf6-6bc849f87c3f)
This works in other scripts just like it is here. But for some reason it doesn't start in the current script.

3) Finally I've got this part that ends the script that I also need to work after the elapsed time:
![image](https://github.com/bonsai-rx/bonsai/assets/48135253/cd0f3ba9-285e-44f2-b4a4-2869f433586c)
This again, works in other cases but not this one. The only difference being the reward line 

Ideally I would want (2) and (1) to start after delays of 5s and 8s and for (3) to happen after a certain time window.
All the rest of the script is various cameras and signals that are not necessarily that important for the current problem, as they generally work in other scripts.

Any ideas why (2) would not work in this instance, and how to make (1) delayed? 

Thanks,

Liad

**Comment by bruno-f-cruz** [comment]
Hi

1. Should be fixed by using 'DelaySubscription' as in your number 2.

2. Do you have a shader context running in your workflow (ie do you have a CreateWindow operator?) shader schedulers only work if you have a shader's context. If you don't want to use that schedule use the Reactive version of the same operator that uses the OS default scheduler.

3. Same as 1. and 2.

**Reply by liadJB** [reply]
Great. Thanks. I do not have a window so that would explain it. 

for 1) where should I place the reactive DelaySubscription? I don't want it to be delayed every repeat but for the entire operation to be delayed once.

Thanks

**Reply by bruno-f-cruz** [reply]
It works the same way as your 2. snippet. If you don't want to delay every repeat, delay the subscription to the full second by putting it at the end of the branch. Documentation can be found here https://bonsai-rx.org/docs/api/Bonsai.Reactive.DelaySubscription.html
